Add support for Hilla

platosha opened the pull request:
Hello! Let me kindly ask for adding the Hilla entry again, now that we have added support for Gradle in the Hilla 2.1 feature release.

This PR is a follow-up for #1113, which was reverted previously. Also see the related entry proposal: #872

pivotal-cla commented:
@platosha Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-io/start.spring.io&pullRequestId=1224)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-io/start.spring.io&pullRequestId=1224) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-cla commented:
@platosha Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-io/start.spring.io&pullRequestId=1224)!

snicoll commented on start-client/dev/api.mock.json:
```diff
@@ -176,6 +176,18 @@
               },
               "reference": { "href": "https://vaadin.com/docs" }
             }
+          },
```
This is a mock file for testing and doesn't need to be updated.

snicoll commented on start-site/src/main/resources/application.yml:
```diff
@@ -163,6 +163,13 @@ initializr:
             version: 24.0.7
           - compatibilityRange: "[3.1.0-M1,3.2.0-M1)"
             version: 24.1.0
+      hilla:
```
BOMs are ordered alphabetically.

snicoll reviewed (commented):
Thanks for the PR. I would need to play with this a bit more but I've added a few comments for your consideration.

platosha commented:
> Thanks for the PR. I would need to play with this a bit more but I've added a few comments for your consideration.

Thank you. Addressed the comments so far.

snicoll commented:
@platosha can you please squash all that and rebase on top of `main` please?

platosha commented:
@snicoll sure, done so
